eureka2-core/src/main/java/com/netflix/eureka2/interests/Index.java

```diff
@@ -77,20 +78,20 @@ public void call(Subscriber<? super ChangeNotification<T>> subscriber) {
                 // Because new instance holder updates will be added while we replay them, they will be
                 // partially visible by the subscriber. When we replay buffered real time updates,
                 // they may overlap with what was already sent from the init holder.
-                NotificationsSubject<T> realTimeSubject = NotificationsSubject.create();
+                // TODO can we make this buffering cheaper?
+                final NotificationsSubject<T> realTimeSubject = NotificationsSubject.create();
                 realTimeSubject.pause();
                 realTimeSource.subscribe(realTimeSubject);
-                realTimeSubject.subscribe(subscriber);
 
-                // Reply initial state
-                for (ChangeNotification<T> notification : initStateHolder) {
-                    subscriber.onNext(notification);
-                }
-
-                // Now open real time source
-                realTimeSubject.resume();
+                realTimeSubject.mergeWith(Observable.from(initStateHolder).doOnCompleted(new Action0() {
```

Comment:
I see. Since, `NotificationSubject` was written before backpressure in RxJava, it may be a good time now to have a second look at it.
I can see this being implemented as an operator that does not request upstream till the init state finish, which naturally buffers in the subject.